src/Writer/MSSQL.php

```diff
@@ -143,13 +150,19 @@ public function write(CsvFile $csv, array $table): void
         $this->logger->info('BCP import finished');
     }
 
-    public function bcpCast(string $srcColName, string $type, string $size, string $colName, int $version): string
-    {
-        if ($version > 10) {
-            return sprintf('TRY_CAST(%s AS %s%s) as %s', $srcColName, $type, $size, $colName);
-        }
-
-        return sprintf('CAST(%s AS %s%s) as %s', $srcColName, $type, $size, $colName);
+    public function bcpCast(
+        string $srcColName,
+        string $type,
+        string $size,
+        string $colName,
+        bool $nullable,
+        int $version
+    ): string {
+        $castFunction = $version > 10 ? 'TRY_CAST' : 'CAST';
+
+        // Null must be converted to empty string, because TRY_CAST(NULL as VARCHAR(255)) is NULL
+        $rawValue = !$nullable && in_array($type, self::$stringTypes) ? "COALESCE($srcColName, '')" : $srcColName;
```

Comment:
Fix spociva v pridani `COALESCE`, tj. konverzuje NULL na prazdny retazec -> ak je stlpec `NOT NULL` a je to niektory z retazcovych typov.